Metaverse/Source/Metaverse/MetaverseCharacter.cpp

```diff
@@ -9,113 +9,204 @@
+    CameraBoom->SetupAttachment(RootComponent);
+    CameraBoom->TargetArmLength =
+        400.0f; // The camera follows at this distance behind the character
+    CameraBoom->bUsePawnControlRotation =
+        true; // Rotate the arm based on the controller
+
+    // Create a follow camera
+    FollowCamera =
+        CreateDefaultSubobject<UCameraComponent>(TEXT("FollowCamera"));
+    FollowCamera->SetupAttachment(
+        CameraBoom,
+        USpringArmComponent::SocketName); // Attach the camera to the end of the
+                                          // boom and let the boom adjust to
+                                          // match the controller orientation
+    FollowCamera->bUsePawnControlRotation =
+        false; // Camera does not rotate relative to arm
+
+    AttributeComponent =
+        CreateDefaultSubobject<UAttributeComponent>(TEXT("Attribute"));
+
+    DefiWalletCoreComponent =
+        CreateDefaultSubobject<UDefiWalletCoreComponent>(TEXT("Defi Wallet Core"));
+
+    // Note: The skeletal mesh and anim blueprint references on the Mesh
+    // component (inherited from Character) are set in the derived blueprint
+    // asset named ThirdPersonCharacter (to avoid direct content references in
+    // C++)
 }
 
-void AMetaverseCharacter::BeginPlay()
-{
-	// Call the base class
-	Super::BeginPlay();
-
-	//Add Input Mapping Context
-	if (APlayerController* PlayerController = Cast<APlayerController>(Controller))
-	{
-		if (UEnhancedInputLocalPlayerSubsystem* Subsystem = ULocalPlayer::GetSubsystem<UEnhancedInputLocalPlayerSubsystem>(PlayerController->GetLocalPlayer()))
-		{
-			Subsystem->AddMappingContext(DefaultMappingContext, 0);
-		}
-	}
+void AMetaverseCharacter::BeginPlay() {
+    // Call the base class
+    Super::BeginPlay();
+
+    // Add Input Mapping Context
+    if (APlayerController *PlayerController =
+            Cast<APlayerController>(Controller)) {
+        if (UEnhancedInputLocalPlayerSubsystem *Subsystem =
+                ULocalPlayer::GetSubsystem<UEnhancedInputLocalPlayerSubsystem>(
+                    PlayerController->GetLocalPlayer())) {
+            Subsystem->AddMappingContext(DefaultMappingContext, 0);
+        }
+    }
+    InitializeMetaverseOverlay();
+    DefiWalletCoreComponent->Setup();
 }
 
 // Input
-void AMetaverseCharacter::SetupPlayerInputComponent(class UInputComponent* PlayerInputComponent)
-{
-	// Set up action bindings
-	if (UEnhancedInputComponent* EnhancedInputComponent = CastChecked<UEnhancedInputComponent>(PlayerInputComponent)) {
+void AMetaverseCharacter::SetupPlayerInputComponent(
+    class UInputComponent *PlayerInputComponent) {
+    // Set up action bindings
+    if (UEnhancedInputComponent *EnhancedInputComponent =
+            CastChecked<UEnhancedInputComponent>(PlayerInputComponent)) {
+
+        // Jumping
+        EnhancedInputComponent->BindAction(JumpAction, ETriggerEvent::Triggered,
+                                           this, &ACharacter::Jump);
```

Comment:
indentation here and below?

Reply:
The indentation is formatted by clang-format (I found auto formatting is not very stable - sometimes, after formatting, the MACRO did not work, I will format it in another PR, as long as it works now), the original file created by unreal template is unformatted.. Any issues?